src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyOperationManager.cs

```diff
@@ -138,6 +140,11 @@ public virtual bool SetupChannel(IEnumerable<string> sources)
                 catch (Exception ex)
                 {
                     EqtTrace.Error("ProxyOperationManager: Failed to launch testhost :{0}", ex);
+
+                    if (ex is AggregateException && ex.InnerException is TaskCanceledException)
+                    {
+                        throw new TestPlatformException(Common.Resources.Resources.CancellationRequested);
```

Comment:
TaskCanceledException might happen because of something other than cancel also. In that case, we might not want exception to part of TestPlatformException.

Here in case of TaskCancelledException, we can check `this.CancellationTokenSource.Token.ThrowTestPlatformExceptionIfCancellationRequested();` which will throw TestPlatformException only in case of cancellation.

Reply:
Makes sense. Done